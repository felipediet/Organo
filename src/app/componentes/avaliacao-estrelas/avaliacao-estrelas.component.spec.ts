import "zone.js";
import "zone.js/testing";
import { ComponentFixture, TestBed } from "@angular/core/testing";
import { AvaliacaoEstrelasComponent } from "./avaliacao-estrelas.component";
import { BrowserDynamicTestingModule, platformBrowserDynamicTesting } from "@angular/platform-browser-dynamic/testing";
import { NG_VALUE_ACCESSOR } from "@angular/forms";
import { forwardRef } from "@angular/core";

describe('AvaliacaoEstrelasComponent', () => {	
    let component: AvaliacaoEstrelasComponent;
    let fixture: ComponentFixture<AvaliacaoEstrelasComponent>;

    // beforeAll(() => {
    //     TestBed.initTestEnvironment(BrowserDynamicTestingModule, platformBrowserDynamicTesting());
    // });
    
    beforeEach(() => {
        TestBed.configureTestingModule({
        imports: [AvaliacaoEstrelasComponent],
        providers: [
            {
            provide: NG_VALUE_ACCESSOR,
            useExisting: forwardRef(() => AvaliacaoEstrelasComponent),
            multi: true
            }
        ]
        });

        fixture = TestBed.createComponent(AvaliacaoEstrelasComponent);
        component = fixture.componentInstance;
        component.readOnly = false;
    });

    it('deveria ser criado', () => {
        expect(component).toBeTruthy();
    });

    it('deveria definir a propriedade classificacao', () => {
        expect(component.classificacao).toBeDefined();
    });

    it('deveria renderizar o conteúdo baseado na propriedade classificacao', () => {
        component.classificacao = 3;
        expect(component).toMatchSnapshot();
    });

    it('deveria renderizar o conteúdo baseado na propriedade readOnly', () => {
        component.readOnly = true;
        expect(component).toMatchSnapshot();
    });

    it('deveria renderizar o conteúdo baseado na propriedade estrelas', () => {
        expect(component).toMatchSnapshot();
    });

    it('deveria atribuir um valor para a calssificação quando o método writeValue for chamado', () => {
        component.writeValue(3);
        expect(component.classificacao).toBe(3);
    });



});