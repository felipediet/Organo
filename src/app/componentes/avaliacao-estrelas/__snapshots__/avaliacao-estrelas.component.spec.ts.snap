// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvaliacaoEstrelasComponent deveria renderizar o conteúdo baseado na propriedade classificacao 1`] = `
AvaliacaoEstrelasComponent {
  "__ngContext__": 4,
  "classificacao": 3,
  "estrelas": [
    1,
    2,
    3,
    4,
    5,
  ],
  "onChange": [Function],
  "onTouched": [Function],
  "readOnly": false,
}
`;

exports[`AvaliacaoEstrelasComponent deveria renderizar o conteúdo baseado na propriedade estrelas 1`] = `
AvaliacaoEstrelasComponent {
  "__ngContext__": 8,
  "classificacao": 1,
  "estrelas": [
    1,
    2,
    3,
    4,
    5,
  ],
  "onChange": [Function],
  "onTouched": [Function],
  "readOnly": false,
}
`;

exports[`AvaliacaoEstrelasComponent deveria renderizar o conteúdo baseado na propriedade readOnly 1`] = `
AvaliacaoEstrelasComponent {
  "__ngContext__": 6,
  "classificacao": 1,
  "estrelas": [
    1,
    2,
    3,
    4,
    5,
  ],
  "onChange": [Function],
  "onTouched": [Function],
  "readOnly": true,
}
`;
